### ISSUE TITLE
Pyright reports incorrect reportReturnType for `Protocol`'s method with docstring

### ISSUE THREAD

**Issue by xymy** [microsoft/pyright#9037]
**Describe the bug**

The recent pyright reports a weird error for `Protocol`'s method with docstring. See the code below. Looks like pyright expects a `...` at the end of method. But previous version of pyright does not have this weird behavior.

**Code or Screenshots**

```python
from typing import Protocol


# mypy and pyright passed
class Handler1(Protocol):
    def process(self) -> str: ...


# mypy passed but pyright said "None" is not assignable to "str" Pylance(reportReturnType)
class Handler2(Protocol):
    def process(self) -> str:
        """Process the event."""


# mypy and pyright passed
class Handler3(Protocol):
    def process(self) -> str:
        """Process the event."""
        ...
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

I used Pylance v2024.9.1 in VS Code 1.93.1

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

You mentioned that you are seeing a change in behavior in recent versions of pyright, but its behavior hasn't changed in this regard for a long time, if ever. I confirmed that it produces the same error as far back as 1.1.332, which is the oldest version currently supported in the [pyright playground](https://pyright-play.net/). That's over a year old.

The Python Typing Spec indicates that type checkers should determine when a protocol provides a default implementation for a method. See [this section](https://typing.readthedocs.io/en/latest/spec/protocol.html#explicitly-declaring-implementation). The spec does not currently provide details about what should be considered an implementation and what should be considered a non-implementation. There's general agreement among tools that a `...` or a `raise NotImplementedError()` alone or with a docstring means "not implemented". There is not agreement among tools how a docstring by itself should be interpreted. Pyright treats this as an implemented function and type checks it accordingly. 

Unless / until the typing spec is updated to indicate that this behavior is incorrect, I'm going to leave pyright's behavior as is. Changing it could be potentially disruptive to pyright users who rely on the current behavior.

If you want your protocols to work with pyright, add a `...` to clearly indicate that the method is not meant to be a default implementation.
